include/LightGBM/utils/openmp_wrapper.h

```diff
@@ -4,6 +4,21 @@
  */
 #ifndef LIGHTGBM_OPENMP_WRAPPER_H_
 #define LIGHTGBM_OPENMP_WRAPPER_H_
+
+/*
+ * See https://github.com/dmlc/dmlc-core/blob/main/include/dmlc/omp.h#L14
+ */
+#if defined(__clang__)
+#undef __GOMP_NOTHROW
+#define __GOMP_NOTHROW
+#elif defined(__cplusplus)
+#undef __GOMP_NOTHROW
+#define __GOMP_NOTHROW throw()
+#else
+#undef __GOMP_NOTHROW
+#define __GOMP_NOTHROW __attribute__((__nothrow__))
+#endif
```

Comment:
I guess we don't need these `define`s in case of OpenMP is found and they should go under `#else` branch of `#ifdef _OPENMP` condition (line 67), shouldn't they?

Reply:
Make sense / I just moved it over there.